pulsar-common/src/main/proto/PulsarApi.proto

```diff
@@ -62,9 +62,9 @@ message MessageIdData {
     optional int32 batch_index = 4 [default = -1];
     repeated int64 ack_set = 5;
     optional int32 batch_size = 6;
-
     // For the chunk message id, we need to specify the first chunk message id.
     optional MessageIdData first_chunk_message_id = 7;
+    optional string topicName = 8;
```

Comment:
I think we need a PIP to add new fields to the Pulsar API. In [PIP-224](https://github.com/apache/pulsar/issues/18616) I proposed to add an extra class `TopicMessageIdSerDes` for the serialization of the `TopicMessageId`, but the related PR is not present yet. If we can introduce such a new field, I think we don't need to implement `TopicMessageIdSerDes` any more.

Reply:
As I described [here](https://github.com/apache/pulsar/issues/18616#issuecomment-1328617005), the limit of PIP-224 is caused by the lack of the topic name field in `MessageIdData`. I think it would be better to add this field before the 3.0.0 release. But I'd like to hear more voices about this change.

Reply:
@BewareMyPower can you please unblock the PR as I have many times mentioned in many other PRs where few committers block all the PRs without any reason and miss or don't unblock PRs even after addressing the all concerns. I don't want to do such destructive stuff but please don't block PRs unnecessarily.

Reply:
Actually I'm +1 to your PR and I think the API proposed in PIP-224 could be replaced with this PR.

> few committers block all the PRs without any reason

However, the reason I blocked this PR is the current requirement of the https://github.com/apache/pulsar/blob/master/wiki/proposals/PIP.md

> Any change to the wire protocol APIs

This PR brings a change to the wire protocol API, and I think it should not be passed without any proposal. /cc @merlimat @eolivelli @codelipenghui

Reply:
Hmmm, I think it's better to add a description to clarify the new field will not apply to RPC calls. It is only used by the client side to carry the topic name.